### PR TITLE
[magnum-*] Remove accidentally generated parameter and optimized portfile.cmake

### LINKED ISSUE #24710: [magnum-*] Features not available
A call to `vcpkg_check_features` is missing.

### PR DESCRIPTION
**Pull request by Cheney-W**
**Describe the pull request**

- #### What does your PR fix?
  Fixes https://github.com/microsoft/vcpkg/issues/24710

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Review by LilyWangLL — DISMISSED** [review]

File: ports/magnum-extras/portfile.cmake [hunk 1]
```diff
@@ -11,32 +11,33 @@ string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC)
 # Handle features
 set(_COMPONENT_FLAGS "")
```
Comment (BillyONeal) [inline ports/magnum-extras/portfile.cmake:12]: There is literally 1 feature in this. I think it should use `vcpkg_check_features` instead.

File: ports/magnum-integration/portfile.cmake [hunk 2]
```diff
@@ -12,54 +12,57 @@ string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC)
 set(_COMPONENT_FLAGS "")
 foreach(_feature IN LISTS ALL_FEATURES)
```
Comment (BillyONeal) [inline ports/magnum-integration/portfile.cmake:13]: Ditto this block is much longer than a `vcpkg_check_features`

**Review by BillyONeal — CHANGES_REQUESTED** [review]
I think this should do:

```diff
diff --git a/ports/magnum-extras/portfile.cmake b/ports/magnum-extras/portfile.cmake
index 54933251a..6515e3945 100644
--- a/ports/magnum-extras/portfile.cmake
+++ b/ports/magnum-extras/portfile.cmake
@@ -8,28 +8,16 @@ vcpkg_from_github(
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC)
 
-# Handle features
-set(_COMPONENT_FLAGS "")
-foreach(_feature IN LISTS ALL_FEATURES)
-    if(_feature)
-        # Uppercase the feature name and replace "-" with "_"
-        string(TOUPPER "${_feature}" _FEATURE)
-        string(REPLACE "-" "_" _FEATURE "${_FEATURE}")
-
-        # Turn "-DWITH_*=" ON or OFF depending on whether the feature
-        # is in the list.
-        if(_feature IN_LIST FEATURES)
-            list(APPEND _COMPONENT_FLAGS "-DWITH_${_FEATURE}=ON")
-        else()
-            list(APPEND _COMPONENT_FLAGS "-DWITH_${_FEATURE}=OFF")
-        endif()
-    endif()
-endforeach()
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        ui WITH_UI
+)
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        ${_COMPONENT_FLAGS}
+        ${FEATURE_OPTIONS}
         -DBUILD_STATIC=${BUILD_STATIC}
         -DMAGNUM_PLUGINS_DEBUG_DIR=${CURRENT_INSTALLED_DIR}/debug/bin/magnum-d
         -DMAGNUM_PLUGINS_RELEASE_DIR=${CURRENT_INSTALLED_DIR}/bin/magnum
@@ -40,7 +28,7 @@ vcpkg_cmake_install()
 vcpkg_cmake_config_fixup(PACKAGE_NAME MagnumExtras CONFIG_PATH share/cmake/MagnumExtras)
 
 # Messages to the user
-if("ui" IN_LIST FEATURES)
+if(WITH_UI)
     message(WARNING "It is recommended to install one of magnum-plugins[freetypefont,harfbuzzfont,stbtruetypefont] to have the UI library working out of the box")
 endif()
 
diff --git a/ports/magnum-integration/portfile.cmake b/ports/magnum-integration/portfile.cmake
index a519ee18d..f62ea902b 100644
--- a/ports/magnum-integration/portfile.cmake
+++ b/ports/magnum-integration/portfile.cmake
@@ -8,28 +8,19 @@ vcpkg_from_github(
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" BUILD_STATIC)
 
-# Handle features
-set(_COMPONENT_FLAGS "")
-foreach(_feature IN LISTS ALL_FEATURES)
-    # Uppercase the feature name and replace "-" with "_"
-    if(_feature)
-        string(TOUPPER "${_feature}" _FEATURE)
-        string(REPLACE "-" "_" _FEATURE "${_FEATURE}")
-
-        # Turn "-DWITH_*=" ON or OFF depending on whether the feature
-        # is in the list.
-        if(_feature IN_LIST FEATURES)
-            list(APPEND _COMPONENT_FLAGS "-DWITH_${_FEATURE}=ON")
-        else()
-            list(APPEND _COMPONENT_FLAGS "-DWITH_${_FEATURE}=OFF")
-        endif()
-    endif()
-endforeach()
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        bullet WITH_BULLET
+        eigen WITH_EIGEN
+        glm WITH_GLM
+        imgui WITH_IMGUI
+)
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        ${_COMPONENT_FLAGS}
+        ${FEATURE_OPTIONS}
         -DBUILD_STATIC=${BUILD_STATIC}
         -DMAGNUM_PLUGINS_DEBUG_DIR=${CURRENT_INSTALLED_DIR}/debug/bin/magnum-d
         -DMAGNUM_PLUGINS_RELEASE_DIR=${CURRENT_INSTALLED_DIR}/bin/magnum
```

File: ports/magnum-extras/portfile.cmake (hunk 1 above)
Comment (Cheney-W) [reply ports/magnum-extras/portfile.cmake]: Thanks for your suggestion.

File: ports/magnum-integration/portfile.cmake (hunk 2 above)
Comment (Cheney-W) [reply ports/magnum-integration/portfile.cmake]: Thanks!

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thank you!

**Pull request merged**